File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestFloat.js
print("getFloat")
if(typeof getFloat !== 'undefined')
	throw "getFloat already defined!"
getFloat = callback("getFloat")
getFloatType = typeof getFloat
print(getFloatType)
print(getFloat)
if(getFloatType !== 'number')
	throw "getFloat not number!"
callback("setFloat",getFloat)

print("getFloatVector")
if(typeof getFloatVector !== 'undefined')
	throw "getFloatVector already defined!"
getFloatVector = callback("getFloatVector")
getFloatVectorType = typeof getFloatVector[0]
print(getFloatVectorType)
print(getFloatVector)
if(getFloatVectorType !== 'number')
	throw "getFloatVector not number!"
callback("setFloatVector",getFloatVector)

print("getFloatVectorAsList")
if(typeof getFloatVectorAsList !== 'undefined')
	throw "getFloatVectorAsList already defined!"
getFloatVectorAsList = callback("getFloatVectorAsList")
getFloatVectorAsListType = typeof getFloatVectorAsList[0]
print(getFloatVectorAsListType)
print(getFloatVectorAsList)
if(getFloatVectorAsListType !== 'number')
	throw "getFloatVectorAsList not number!"
callback("setFloatVectorAsList",getFloatVectorAsList)

print("getFloatMatrix")
if(typeof getFloatMatrix !== 'undefined')
	throw "getFloatMatrix already defined!"
getFloatMatrix = callback("getFloatMatrix")
getFloatMatrixType = typeof getFloatMatrix[0][0]
print(getFloatMatrixType)
print(getFloatMatrix)
if(getFloatMatrixType !== 'number')
	throw "getFloatMatrix not number!"
callback("setFloatMatrix",getFloatMatrix)

print("getFloatMatrixAsList")
if(typeof getFloatMatrixAsList !== 'undefined')
	throw "getFloatMatrixAsList already defined!"
getFloatMatrixAsList = callback("getFloatMatrixAsList")
getFloatMatrixAsListType = typeof getFloatMatrixAsList[0][0]
print(getFloatMatrixAsListType)
print(getFloatMatrixAsList)
if(getFloatMatrixAsListType !== 'number')
	throw "getFloatMatrixAsList not number!"
callback("setFloatMatrixAsList",getFloatMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-ruby-contract/src/test/java/de/invesdwin/context/ruby/tests/callback/ParametersAndReturnsTestFloat.rb
puts("getFloat")
unless (defined?(getFloat)).nil? || getFloat.nil?
	raise "getFloat already defined!"
end
getFloat = callback("getFloat")
getFloatType = getFloat.class
puts(getFloatType)
puts(getFloat)
if(getFloatType != Float)
	raise "getFloat not Float!"
end
callback("setFloat", getFloat)

puts("getFloatVector")
unless (defined?(getFloatVector)).nil? || getFloatVector.nil?
	raise "getFloatVector already defined!"
end
getFloatVector = callback("getFloatVector")
getFloatVectorType = getFloatVector[0].class
puts(getFloatVectorType)
puts(getFloatVector)
if(getFloatVectorType != Float)
	raise "getFloatVector not Float!"
end
callback("setFloatVector", getFloatVector)

puts("getFloatVectorAsList")
unless (defined?(getFloatVectorAsList)).nil? || getFloatVectorAsList.nil?
	raise "getFloatVectorAsList already defined!"
end
getFloatVectorAsList = callback("getFloatVectorAsList")
getFloatVectorAsListType = getFloatVectorAsList[0].class
puts(getFloatVectorAsListType)
puts(getFloatVectorAsList)
if(getFloatVectorAsListType != Float)
	raise "getFloatVectorAsList not Float!"
end
callback("setFloatVectorAsList", getFloatVectorAsList)

puts("getFloatMatrix")
unless (defined?(getFloatMatrix)).nil? || getFloatMatrix.nil?
	raise "getFloatMatrix already defined!"
end
getFloatMatrix = callback("getFloatMatrix")
getFloatMatrixType = getFloatMatrix[0][0].class
puts(getFloatMatrixType)
puts(getFloatMatrix)
if(getFloatMatrixType != Float)
	raise "getFloatMatrix not Float!"
end
callback("setFloatMatrix", getFloatMatrix)

puts("getFloatMatrixAsList")
unless (defined?(getFloatMatrixAsList)).nil? || getFloatMatrixAsList.nil?
	raise "getFloatMatrixAsList already defined!"
end
getFloatMatrixAsList = callback("getFloatMatrixAsList")
getFloatMatrixAsListType = getFloatMatrixAsList[0][0].class
puts(getFloatMatrixAsListType)
puts(getFloatMatrixAsList)
if(getFloatMatrixAsListType != Float)
	raise "getFloatMatrixAsList not Float!"
end
callback("setFloatMatrixAsList", getFloatMatrixAsList)
